mangaki/mangaki/algo/sgd.py

```diff
@@ -0,0 +1,66 @@
+import numpy as np
+from scipy.sparse import coo_matrix
+from sklearn.metrics import mean_squared_error
+from sklearn.metrics.pairwise import cosine_similarity
+import random
+
+from mangaki.algo.recommendation_algorithm import RecommendationAlgorithm, register_algorithm
+
+
+@register_algorithm('sgd')
+class MangakiSGD(RecommendationAlgorithm):
```

Comment:
I'd prefer we avoid putting `nb_users` and `nb_works` in the constructor, it is not matching the current logic of all recommendation algorithm, until we move away from `set_params`, we should keep it as-is.

Reply:
Sorry, it was some leftover of an experiment. Removed.

Reply:
Also the reason was: it was better to initialize the parameters in `__init__`, but for this we need the `nb_users` and `nb_works` before it is specified using `set_params`.